src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -53,80 +55,116 @@
+	protected Entity[] get(Event e, LivingEntity[] source) {
 		return get(source, new Converter<LivingEntity, Entity>() {
 			@Override
 			@Nullable
-			public Entity convert(final LivingEntity en) {
+			public Entity convert(LivingEntity en) {
 				if (getTime() >= 0 && e instanceof EntityTargetEvent && en.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
-					final Entity t = ((EntityTargetEvent) e).getTarget();
-					if (t == null || type != null && !type.isInstance(t))
+					Entity target = ((EntityTargetEvent) e).getTarget();
+					if (target == null || type != null && !type.isInstance(target))
 						return null;
-					return t;
+					return target;
 				}
-				return Utils.getTarget(en, type);
+				return getTarget(en, type);
 			}
 		});
 	}
-	
+
 	@Override
 	public Class<? extends Entity> getReturnType() {
 		return type != null ? type.getType() : Entity.class;
 	}
-	
+
 	@Override
-	public String toString(final @Nullable Event e, final boolean debug) {
+	public String toString(@Nullable Event e, boolean debug) {
 		if (e == null)
 			return "the target" + (type == null ? "" : "ed " + type) + (getExpr().isDefault() ? "" : " of " + getExpr().toString(e, debug));
 		return Classes.getDebugMessage(getAll(e));
 	}
-	
+
 	@Override
-	public boolean setTime(final int time) {
+	public boolean setTime(int time) {
 		return super.setTime(time, EntityTargetEvent.class, getExpr());
 	}
-	
+
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
+	public Class<?>[] acceptChange(ChangeMode mode) {
 		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
 			return CollectionUtils.array(LivingEntity.class);
 		return super.acceptChange(mode);
 	}
-	
+
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
+	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
 		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE) {
-			final LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
-			for (final LivingEntity entity : getExpr().getArray(e)) {
+			LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
+			for (LivingEntity entity : getExpr().getArray(e)) {
 				if (getTime() >= 0 && e instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
 					((EntityTargetEvent) e).setTarget(target);
-				} else {
-					if (entity instanceof Creature)
-						((Creature) entity).setTarget(target);
+				} else if (entity instanceof Mob) {
+					((Mob) entity).setTarget(target);
 				}
 			}
-		} else {
-			super.change(e, delta, mode);
+			return;
 		}
+		super.change(e, delta, mode);
 	}
-	
+
+	/**
+	 * Gets an entity's target.
+	 * 
+	 * @param entity The entity to get the target of
+	 * @param type Can be null for any entity
+	 * @return The entity's target
+	 */
+	//TODO Switch this over to RayTraceResults 1.13+ when 1.12 support is dropped.
```

Comment:
You can do this now if you want

Reply:
It's not working properly. Don't know what's wrong with it, so not from me.
```java
RayTraceResult result = entity.getWorld().rayTraceEntities(entity.getEyeLocation(), entity.getLocation().getDirection().normalize(), SkriptConfig.maxTargetBlockDistance.value().longValue(), e -> type.isInstance(e));
    return (T) result.getHitEntity();
``